iexec-sms-library/src/main/java/com/iexec/sms/api/SmsClientProvider.java

```diff
@@ -0,0 +1,58 @@
+package com.iexec.sms.api;
+
+import com.iexec.common.chain.IexecHubAbstractService;
+import com.iexec.common.task.TaskDescription;
+import feign.Logger;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Manages the {@link SmsClient}, providing an easy way to access SMS
+ * and avoiding the need to create a new {@link SmsClient} instance each time.
+ */
+@Slf4j
+public class SmsClientProvider {
+    private final Map<String, SmsClient> urlToSmsClient = new HashMap<>();
```

Comment:
Don't you think entries should expire at some point?

Reply:
We should have only a small set of SMS, so this map should never have a big memory footprint. Using an expiring map could lead to way more memory & CPU consumption than simply keeping the entries there. Don't you think so?

Reply:
I agree, I read `taskIdToSmsClient` I was concerned about keys, thanks!